pkg/transparentproxy/validate/validator_test.go

```diff
@@ -64,11 +67,16 @@ var _ = Describe("Should Validate iptables rules", func() {
 		Expect(err).To(HaveOccurred())
 		errMsg := err.Error()
 		containsTimeout := strings.Contains(errMsg, "i/o timeout")
-		containsRefused := strings.Contains(errMsg, "connection refused")
+		containsRefused := strings.Contains(errMsg, "refused")
 		Expect(containsTimeout || containsRefused).To(BeTrue())
 	})
 })
 
-func createValidator(ipv6Enabled bool) *Validator {
-	return NewValidator(ipv6Enabled, ValidationServerPort, core.NewLoggerTo(os.Stdout, kuma_log.InfoLevel).WithName("validator"))
+func createValidator(ipv6Enabled bool, validationServerPort uint16) *Validator {
+	return NewValidator(ipv6Enabled, validationServerPort, core.NewLoggerTo(os.Stdout, kuma_log.InfoLevel).WithName("validator"))
+}
+
+func randomPort() uint16 {
```

Comment:
We need to check if something is listening on a port you pick. Use `pkg/util/net#PickTCPPort`

Reply:
sorry, it's `pkg/test#GetFreePort`. We should remove the `pkg/util/net#PickTCPPort` util.